Examples/Io/EDM4hep/src/EDM4hepUtil.cpp

```diff
@@ -0,0 +1,305 @@
+      from.getMomentum().x * Acts::UnitConstants::GeV,
+      from.getMomentum().y * Acts::UnitConstants::GeV,
+      from.getMomentum().z * Acts::UnitConstants::GeV,
+  };
+  const auto energy = std::hypot(momentum.norm(), mass);
+
+  ActsFatras::Hit::Vector4 pos4{
+      from.getPosition().x * Acts::UnitConstants::mm,
+      from.getPosition().y * Acts::UnitConstants::mm,
+      from.getPosition().z * Acts::UnitConstants::mm,
+      from.getTime() * Acts::UnitConstants::ns,
+  };
+
+  ActsFatras::Hit::Vector4 mom4{
+      momentum.x(),
+      momentum.y(),
+      momentum.z(),
+      energy,
+  };
+
+  // TODO no EDM4hep equivalent?
+  ActsFatras::Hit::Vector4 delta4{
+      0 * Acts::UnitConstants::GeV, 0 * Acts::UnitConstants::GeV,
+      0 * Acts::UnitConstants::GeV,
+      0 * Acts::UnitConstants::GeV,  // sth.getEDep()
+  };
+
+  // TODO no EDM4hep equivalent?
+  int32_t index = -1;
+
+  return ActsFatras::Hit(geometryId, particleId, pos4, mom4, mom4 + delta4,
+                         index);
+}
+
+void EDM4hepUtil::writeSimHit(const ActsFatras::Hit& from,
+                              edm4hep::MutableSimTrackerHit to,
+                              MapParticleIdTo particleMapper,
+                              MapGeometryIdTo geometryMapper) {
+  const Acts::Vector4& globalPos4 = from.fourPosition();
+  const Acts::Vector4& momentum4Before = from.momentum4Before();
+  const auto delta4 = from.momentum4After() - momentum4Before;
+
+  if (particleMapper != nullptr) {
+    to.setMCParticle(particleMapper(from.particleId()));
+  }
+
+  if (geometryMapper != nullptr) {
+    // TODO what about the digitization?
+    to.setCellID(geometryMapper(from.geometryId()));
+  }
+
+  to.setTime(globalPos4[Acts::eTime] / Acts::UnitConstants::ns);
+
+  to.setPosition({
+      globalPos4[Acts::ePos0] / Acts::UnitConstants::mm,
+      globalPos4[Acts::ePos1] / Acts::UnitConstants::mm,
+      globalPos4[Acts::ePos2] / Acts::UnitConstants::mm,
+  });
+
+  to.setMomentum({
+      (float)(momentum4Before[Acts::eMom0] / Acts::UnitConstants::GeV),
+      (float)(momentum4Before[Acts::eMom1] / Acts::UnitConstants::GeV),
+      (float)(momentum4Before[Acts::eMom2] / Acts::UnitConstants::GeV),
+  });
+
+  to.setEDep(-delta4[Acts::eEnergy] / Acts::UnitConstants::GeV);
+}
+
+Measurement EDM4hepUtil::readMeasurement(
+    edm4hep::TrackerHitPlane from,
+    const edm4hep::TrackerHitCollection* fromClusters, Cluster* toCluster,
+    MapGeometryIdFrom geometryMapper) {
+  // no need for digitization as we only want to identify the sensor
+  Acts::GeometryIdentifier geometryId = geometryMapper(from.getCellID());
+
+  // TODO what about the hit index?
+  IndexSourceLink sourceLink{geometryId, 0};
+
+  auto pos = from.getPosition();
+  auto cov = from.getCovMatrix();
+
+  DigitizedParameters dParameters;
+
+  dParameters.indices.push_back(Acts::eBoundLoc0);
+  dParameters.values.push_back(pos.x);
+  dParameters.variances.push_back(cov[0]);
+
+  // TODO cut this out for 1D
```

Comment:
Can we identify the dimensionality without checking/enumerating the edm4hep hit types?

Reply:
I fear we can only go through the type or do something like setting U/V to zero